src/middlewared/middlewared/plugins/cloud/crud.py

```diff
@@ -49,11 +49,13 @@ async def _basic_validate(self, verrors, name, data):
         if verrors:
             return
 
-        provider = REMOTES[credentials["provider"]["type"]]
+        cred_type = credentials["provider"]["type"]
+        provider = REMOTES[cred_type]
+        AttributesModel = {"B2": B2TaskAttributes, "DROPBOX": DropboxTaskAttributes}.get(cred_type, CloudTaskAttributes)
```

Comment:
Can we have this mapping defined in the API file? That way we'll have to import less symbols there, and future changes will only require changing one file.